pkg/cmd/build/build.go

```diff
@@ -212,67 +212,172 @@ func translateDockerErr(err error) error {
+		manifestBuildOptions.Namespace = cmdBuildOptions.Namespace
+	}
+	if len(cmdBuildOptions.Secrets) > 0 {
+		manifestBuildOptions.Secrets = cmdBuildOptions.Secrets
+	}
+	if cmdBuildOptions.NoCache != manifestBuildOptions.NoCache {
+		manifestBuildOptions.NoCache = cmdBuildOptions.NoCache
+	}
+	if len(cmdBuildOptions.BuildArgs) > 0 {
+		manifestBuildOptions.BuildArgs = cmdBuildOptions.BuildArgs
+	}
+	if len(cmdBuildOptions.CacheFrom) > 0 {
+		manifestBuildOptions.CacheFrom = cmdBuildOptions.CacheFrom
+	}
+	if cmdBuildOptions.ExportCache != manifestBuildOptions.ExportCache {
+		manifestBuildOptions.ExportCache = cmdBuildOptions.ExportCache
+	}
 
+	// override output mode from cmdBuildOptions
+	if cmdBuildOptions.OutputMode != "" {
+		manifestBuildOptions.OutputMode = setOutputMode(cmdBuildOptions.OutputMode)
+	} else if manifestBuildOptions.OutputMode == "" {
+		manifestBuildOptions.OutputMode = setOutputMode(oktetoLog.GetOutputFormat())
 	}
 
-	file := b.Dockerfile
-	if !filepath.IsAbs(b.Dockerfile) && !model.FileExistsAndNotDir(file) {
-		file = filepath.Join(b.Context, b.Dockerfile)
+	if cmdBuildOptions.Tag != "" {
+		manifestBuildOptions.Tag = cmdBuildOptions.Tag
 	}
-	opts := &BuildOptions{
-		CacheFrom:   b.CacheFrom,
-		Target:      b.Target,
-		Path:        b.Context,
-		Tag:         b.Image,
-		File:        file,
-		BuildArgs:   args,
-		ExportCache: b.ExportCache,
+
+	// override Tag if global registry
+	if okteto.Context().IsOkteto && cmdBuildOptions.BuildToGlobal {
+		manifestBuildOptions.Tag = registry.ReplaceTargetRepository(manifestBuildOptions.Tag, okteto.GlobalRegistry, manifestBuildOptions.Namespace)
 	}
 
-	outputMode := oktetoLog.GetOutputFormat()
-	if o != nil && o.OutputMode != "" {
-		outputMode = o.OutputMode
+	return manifestBuildOptions
+}
+
+// OptsFromManifest returns the parsed options for the build from the manifest
+func OptsFromManifest(service string, manifestBuildInfo *model.BuildInfo, cmdBuildOptions *BuildOptions) *BuildOptions {
+	manifestBuildOptions := getBuildOptionsFromManifest(service, manifestBuildInfo)
+	return overrideManifestBuildOptions(manifestBuildOptions, cmdBuildOptions)
+}
+
+// optimizedGlobalBuild returns if build should be optimized because the image is already at the global registry
+func (opts *BuildOptions) optimizedGlobalBuild() (string, bool, error) {
+	oktetoLog.Debugf("Applying global build optimization for image %s", opts.Tag)
+
+	globalReference := opts.Tag
+	if registry.IsDevRegistry(opts.Tag) || registry.IsExtendedOktetoRegistry(opts.Tag) {
+		globalReference = registry.ReplaceTargetRepository(opts.Tag, okteto.GlobalRegistry, okteto.DefaultGlobalNamespace)
+
 	}
-	opts.OutputMode = setOutputMode(outputMode)
+	tagWithDigest, err := registry.GetImageTagWithDigest(globalReference)
+	if err != nil {
+		return opts.Tag, false, err
+	}
+	oktetoLog.Debugf("skipping build: image %s is already built", opts.Tag)
+	opts.Tag = globalReference
+	return tagWithDigest, true, nil
+}
```

Comment:
1. We could remove the `bool` in the return since it's the same as `err == nil`. 
2. I think this function should be named something like `GetGlobalBuildTag` since it does not optimize the build. I'm still thinking about the `opts.Tag` being set here
3. The `oktetoLog.Debugf` should be on the caller, since it does not optimize the build

Reply:
What do you think about checking also the dev registry in order to merge the logic from `checkImageAtRegistry`?

Reply:
1. yes, i'm rethinking this part
2. yes, i would change this. i would have to assign this new value out of the func if global tag or tag is found....
3. At the initial approach the logic was merged, but i wanted to separate the global check with the dev check, i will give it a thought
thanks!